opengovsg/postmangovsg
# fix: re-fetch campaign details after campaign is sent

## Problem

When a user first sends a campaign, `sentAt` remains `undefined`. This causes the file name of downloaded delivery reports to contain `invalid_Date`. This issue doesn't occur after refreshing the page since the refresh causes `sentAt` to be refetched from the backend.

Closes #1113

## Solution

**Improvements**:

- Refactor the `onModalConfirm` handler to a common utilility function.

**Bug Fixes**:

- Set the `sentAt` field to the current `Date` when sending a campaign. This shouldn't be far off from the actual value in the backend, since the actual value is set when creating sending jobs as part of the API request handler for sending a campaign.
- Correct the type definitions for `sentAt` and `statusUpdatedAt`. These are should be of type `string | undefined`.

## Before & After Screenshots

There are no visual changes.

## Tests

1. Deploy the change on Amplify.
2. Send a campaign and wait for the delivery report to be generated without refreshing the page.
3. Download the delivery report. The filename should represent the same date as the one shown in the "Sent Date" column.
4. Repeat the test for every campaign type (SMS, Telegram, Email, Protected Email). 

## Deploy Notes

There are no deploy notes.
